### ISSUE TITLE
`cast` <> Truffle Dashboard

### ISSUE THREAD

**Issue by pcaversaccio** [foundry-rs/foundry#5107]
### Component

`cast`

### Describe the feature you would like

[Truffle Dashboard](https://trufflesuite.com/docs/truffle/how-to/use-the-truffle-dashboard) is a very convenient tool that lets you deploy and interact without putting your private key into a `.env` file. Unfortunately, currently, it's _not_ possible to interact with the Truffle Dashboard due to an underlying problem with MetaMask:

- https://github.com/MetaMask/metamask-extension/issues/11824
- https://github.com/MetaMask/metamask-extension/issues/18076
- https://github.com/foundry-rs/foundry/issues/3890

_TL;DR:_ MetaMask supports the tx type `0x2` but not `0x02` which in turn makes it incompatible with the `ethers-rs` library.

I have created this issue to track the status of the ongoing issue(s) and once resolved it should be properly tested and validated, and eventually, I propose to add a small section to the docs. I feel like this is an important part to highlight the advantages of non-`.env`-file-based development.

### Additional context

See my tweet [here](https://twitter.com/pcaversaccio/status/1664930332175478784) about why `.env` files are bad practice.

**Comment by zerosnacks** [comment]
Marking as `not planned` given that Truffle development has been sunset

The issues linked have been resolved, no clear actionable items left

Would be interested in brainstorming on the idea of how we could connect Metamask to Foundry. We do have support for hardware wallets and encrypted keystores now.
